fetch_raw_data.py

```diff
@@ -157,17 +157,27 @@ def fetch_from_recovery_csv(user, google_cloud_credentials_file_path, raw_data_d
         log.info(f"Exported TracedData")
 
 def fetch_listening_groups_csvs(google_cloud_credentials_file_path, pipeline_configuration, raw_data_dir):
+
+    uploaded_listening_group_files = [file.split("/")[-1] for file in google_cloud_utils.list_blobs(google_cloud_credentials_file_path,
+                                                                   pipeline_configuration.project_dataset_bucket_url,
+                                                                   pipeline_configuration.listening_group_bucket_path)]
+
     for k, v in pipeline_configuration.listening_group_csv_urls.items():
         for i, listening_group_csv_url in enumerate(v):
-            listening_group = listening_group_csv_url.split("/")[-1]
+            listening_group_csv = listening_group_csv_url.split("/")[-1]
+
+            if listening_group_csv not in uploaded_listening_group_files:
```

Comment:
The pythonic way of doing this would be to try except on each of the files. Then you don't need your new configuration items project_dataset_bucket_url and listening_group_bucket_path :)

(the try except patterns also work better in places where someone else can change the data while you're working on it, e.g. if someone had deleted the lg file while this script was between lines 161 and 169, though I admit that's extremely unlikely in our case!)